src/parser/Parser.cpp

```diff
@@ -1,122 +1,104 @@
 #include <Parser.h>
 
-Parser::Parser(Lexer lexer)
-    : m_lexer(std::move(lexer))
-    , m_currentToken(m_lexer.getToken())
+Parser::Parser(const Lexer lexer)
+    : m_lexer(lexer)
+    , m_tokens()
 {
 
 }
 
-void Parser::buildTree() {
-    parseStatement();
-}
-
-std::unique_ptr<AST::Stmt> Parser::parseStatement() {
-    std::unique_ptr<AST::Stmt> statement;
-    Token identifier;
-    switch(m_currentToken.type) {
-        case TokenType::Def:
-            statement = parseFunction();
-        case TokenType::Return:
-            statement = parseReturn();
-        case TokenType::Identifier:
-            identifier = m_currentToken;
-            getNextToken();
-            switch(m_currentToken.type) {
-                case TokenType::LeftParen:
-                    getNextToken();
-            }
-            break;
-
-        // todo: other statements
-
-        default:
-            statement = parseExpression();
+std::unique_ptr<AST::Stmt> Parser::parse() {
+    Token tok = m_lexer.getToken();
+    if (tok.type == TokenType::EOL || tok.type == TokenType::END) {
+        // TODO what to do here
+        return nullptr;
     }
-    getNextToken();
-    return std::move(statement);
+    m_tokens.push_back(tok);
```

Comment:
what does this do?

Reply:
It's for backtracking but Thiabaud believes that won't be necessary so I'll remove this.